internal/app/machined/pkg/controllers/network/cmdline.go

```diff
@@ -0,0 +1,122 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package network
+
+import (
+	"fmt"
+	"net"
+	"sort"
+	"strings"
+
+	"github.com/talos-systems/go-procfs/procfs"
+	"inet.af/netaddr"
+)
+
+// CmdlineNetworking contains parsed cmdline networking settings.
+type CmdlineNetworking struct {
+	DHCP         bool
+	Address      netaddr.IPPrefix
+	Gateway      netaddr.IP
+	Hostname     string
+	LinkName     string
+	DNSAddresses []netaddr.IP
+	NTPAddresses []netaddr.IP
+}
+
+// ParseCmdlineNetwork parses `ip=` kernel cmdline argument producing all the available configuration options.
+//
+//nolint:gocyclo,cyclop
+func ParseCmdlineNetwork(cmdline *procfs.Cmdline) (settings CmdlineNetworking, err error) {
```

Comment:
No sure if named results are worth it there.
Same for returning a value instead of a pointer.

Reply:
I personally like named results in more complex functions, just my two cents.

Reply:
I don't think returning a pointer is every worth it in general, unless we're returning something huge, as copying is more efficient than passing via the heap (which might be optimized by the Go compiler, but who knows for every invocation?).

As for the named results, I can look into that